eth/fetcher/block_fetcher.go

```diff
@@ -686,7 +686,8 @@ func (f *BlockFetcher) loop() {
 						// Mark the body matched, reassemble if still unknown
 						matched = true
 						if f.getBlock(hash) == nil {
-							block := types.NewBlockWithHeader(announce.header).WithBody(task.transactions[i], task.uncles[i])
+							body := &types.Body{Transactions: task.transactions[i], Uncles: task.uncles[i]}
+							block := types.NewBlockWithHeader(announce.header).WithBody(*body)
```

Comment:
```suggestion
							body := types.Body{Transactions: task.transactions[i], Uncles: task.uncles[i]}
							block := types.NewBlockWithHeader(announce.header).WithBody(body)
```

Why declare it as a pointer only to immediately dereference it?

Reply:
Indeed, no point. Due to the 2nd commit and me overlooking it. Good catch!